tests/logic.tests/logic.test.js

```diff
@@ -79,3 +79,58 @@ describe('createGrid', () => {
     expect(result.grid).toEqual([]);
   });
 });
+
+describe('moveFigureDown', () => {
+  // ['.', '.', 'p'],
+  // ['#', '.', '.'],
+  // ['#', '.', '.'],
+
+  test('should move the figure down if there is space', () => {
+    const initialField = new Field(
+      { rows: 3, cols: 3 },
+      [{ x: 2, y: 0 }],
+      [
+        { x: 0, y: 1 },
+        { x: 0, y: 2 },
+      ]
+    );
+    const newField = moveFigureDown(initialField);
+
+    expect(newField.figurePoints).toEqual([{ x: 2, y: 1 }]);
+  });
+
+  // ['.', '.', 'p'],
+  // ['.', '.', '#'],
+  // ['.', '.', '#'],
+
+  test('should not move the figure down if it overlaps with landscape points', () => {
+    const initialField = new Field(
+      { rows: 3, cols: 3 },
+      [{ x: 2, y: 0 }],
+      [
+        { x: 2, y: 1 },
+        { x: 2, y: 2 },
+      ]
+    );
+    const newField = moveFigureDown(initialField);
+    expect(newField).toEqual(initialField);
+  });
+
+  // ['.', '.', '.'],
+  // ['#', '.', '.'],
+  // ['.', '#', 'p'],
+
+  test('should not move the figure down if it goes out of bounds', () => {
+    const initialField = new Field(
+      { rows: 3, cols: 3 },
+      [{ x: 2, y: 2 }],
+      [
+        { x: 0, y: 1 },
+        { x: 1, y: 2 },
+      ]
+    );
+    const newField = moveFigureDown(initialField);
+
+    expect(newField).toEqual(initialField);
+  });
+});
```

Comment:
U can add a test to check whether the shape stops and does not fall down further:
```
#xxx
#...
##..
```

and it must remain in this position and not fall further
```
#...
#xxx
##..
```
check this case